R/transpose_list.R

```diff
@@ -1,4 +1,12 @@
 #' @useDynLib jsonlite C_transpose_list
 transpose_list <- function(x, names) {
-  .Call(C_transpose_list, x, names)
+  # Sort names before entering C, allowing for a binary search
```

Comment:
Sorting `names` lets us use a binary search in C. Sorting is a lot easier to do in R. If you are willing to add `{stringi}` or `{withr}` as dependencies, we can sort C-style without this ugly `Sys.setlocale()` dance? There might well be other, cleaner ways to do this.